addons/survey/data/mail_template_data.xml

```diff
@@ -62,10 +67,11 @@
             <tr><td style="padding:15px 20px 10px 20px;">
                 <p>Dear <span t-out="object.partner_id.name or 'participant'">participant</span></p>
                 <p>
-                    Here is, in attachment, your certification document for
-                        <strong t-out="object.survey_id.display_name or ''">Feedback Form</strong>
+                    Please find attached your
+                        <strong t-out="object.survey_id.display_name or ''"/>
```

Comment:
I'm not sure what you mean with "demo-filler-data". I also might have misunderstood what was asked.
I did this change because of:

Current:
Dear Billy Fox
Here is, in attachment, your certification document for Furniture Creation Certification
Congratulations for succeeding the test!

To Be
Dear Billy Fox,
Please find attached your SurveyName certification.
Congratulations for passing the test!

Reply:
Actually in
```
Here is, in attachment, your certification document for
<strong t-out="object.survey_id.display_name or ''">Feedback Form</strong>
```
Feedback Form is a filler in case there is no object (for a no-record rendering), allowing to see "how it looks like in real life". You should therefore keep it, like
```
Please find attached your
<strong t-out="object.survey_id.display_name or ''">Furniture Creation</strong>
certification
```
(or anything better than Furniture Creation, used as a demo data / filler data)